rj_gameplay/rj_gameplay/skill/receive.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+
+import rj_gameplay.eval as eval
+import argparse
+import py_trees
+import sys
+import time
+import numpy as np
+from typing import Optional
+
+import stp.skill as skill
+import stp.role as role
+import stp.action as action
+from rj_gameplay.action import receive, capture
+from stp.skill.action_behavior import ActionBehavior
+from stp.skill.rj_sequence import RjSequence as Sequence
+import stp.rc as rc
+from rj_msgs import msg
+
+class IReceive(skill.ISkill, ABC):
+    ...
+
+
+"""
+A skill version of receive so that actions don't have to be called in tactics
+"""
+class Receive(IReceive):
+    
+    def __init__(self,
+            robot:rc.Robot = None):
+
+        self.robot = robot
+        if self.robot is not None:
+            self.receive = receive.Receive(self.robot.id)
+            self.capture = capture.Capture(self.robot.id)
+        else:
+            self.receive = receive.Receive(self.robot)
+            self.capture = capture.Capture()
```

Comment:
I'm confused by these constructors here. Why do we pass in `self.robot` for `Receive`? Isn't the first argument supposed to be an ID?

Reply:
self.robot is None is the else block so it works, I agree that it's confusing tho